utils.py

```diff
@@ -0,0 +1,26 @@
+"""
+Utils functions to reuse in different parts of the codebase
+"""
+import os
+
+from git.cmd import Git
+
+
+def find_files(path_to_folder_from_project_root=".", file_extension=None):
```

Comment:
OK, last comment :)
I'd propose adding a unit test for `find_files` - the point is that in case `find_files` starts to return an empty set of paths, all our existing CI tests will... succeed as they all iterate over files returned by `find_files`. So, perhaps let's just check if it finds `.git/config` or `README.md`?

Reply:
or maybe an assert? assert that .git/config exists in the list. The test will not save us from incorrect use of find_files.

Reply:
well, interestingly actually something has changed in this PR which makes the "selftest" job execute more tests. Here is a snippet from this PR's CI:
```
Run pytest -We .
============================= test session starts =============================
platform win32 -- Python 3.11.6, pytest-7.4.2, pluggy-1.3.0
rootdir: D:\a\devops_tests\devops_tests
collected 15 items

test_notebooks.py ...                                                    [ 20%]
test_todos_annotated.py ........s...                                     [100%]

======================== 14 passed, 1 skipped in 7.77s ========================
```
and here's an equivalent from the main branch (https://github.com/open-atmos/devops_tests/actions/runs/5911809701/job/16034911272):
```
Run pytest -We .
============================= test session starts =============================
platform win32 -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
rootdir: D:\a\devops_tests\devops_tests
collected 13 items

test_notebooks.py ...                                                    [ 23%]
test_todos_annotated.py .......s..                                       [100%]

======================== 12 passed, 1 skipped in 7.77s ========================
```
So somehow, `find_files` produces two more test items.... any idea? (I guess this supports the need for a unit test of `find_files`... and of course no unit test will ever save us from incorrect use of the tested code)

Reply:
probably that is because we add `__init__.py` and `utils.py`?

Reply:
yes, all files count as separate test cases

Reply:
I think this is not a problem that we additionally include `__init__.py`